src/main/kotlin/org/rust/ide/refactoring/inlineFunction/RsInlineFunctionProcessor.kt

```diff
@@ -209,7 +217,50 @@ class RsInlineFunctionProcessor(
         }
     }
 
-    private fun addFunctionBodyToCallerWithCorrectSpacing(leftSibling: PsiElement?, rightSibling: PsiElement?, blockParent: RsBlock, body: RsBlock, childContainingEnclosingStatement: PsiElement?) {
+    private fun replaceParameterUsages(functionDup: RsFunction, caller: PsiElement, funcScope: LocalSearchScope) {
+        val parameterBindingsList = functionDup.valueParameters
+            .filter { it !is RsSelfParameter }
+            .map { it.pat }
+            .filterIsInstance(RsPat::class.java)
+
+        val valueArgumentList = when (caller) {
+            is RsCallExpr -> caller.valueArgumentList
+            is RsDotExpr -> caller.methodCall?.valueArgumentList ?: return
+            else -> return
+        }
+        for ((index, parameter) in parameterBindingsList.withIndex()) {
+            var argument = valueArgumentList.exprList[index]
+            if (argument is RsUnaryExpr) {
+                argument = argument.expr
+            }
+            replaceParameterWithLetDeclaration(parameter, argument, caller)
+        }
+    }
+
+    private fun replaceParameterWithLetDeclaration(parameter: RsPat, argument: RsExpr, caller: PsiElement) {
+        val name: String = parameter.text.removePrefix("mut ")
+        if (name != argument.text) {
```

Comment:
What does this condition do? Seems that it's not required. Argument probably won't be equal to e.g. `a: i32`.

Reply:
It is equal in about half the tests, that *if* statement prevents us from turning this
```rust
fn main() {
    let bar = 10i32;
    foo(bar);
}
```
into this
```rust
fn main() {
    let bar = 10i32;
    let bar = bar;
    println!("{}", bar);
}
```
and creating a *let* declaration where the exact same one exists.

Reply:
Sorry, I didn't notice that it compares only to the name of the parameter. Let's keep it then and see if it works in the wild.